Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -672,20 +672,21 @@ static inline UGPRPair doWasmCall(Register* partiallyConstructedCalleeFrame, JSW
     CodePtr<WasmEntryPtrTag> codePtr;
     EncodedJSValue boxedCallee = CalleeBits::encodeNullCallee();
 
+    Register& calleeStackSlot = partiallyConstructedCalleeFrame[static_cast<int>(CallFrameSlot::callee)];
+
     if (functionIndex < importFunctionCount) {
         JSWebAssemblyInstance::ImportFunctionInfo* functionInfo = instance->importFunctionInfo(functionIndex);
         codePtr = functionInfo->importFunctionStub;
+        calleeStackSlot = -(functionIndex + 1);
     } else {
         // Target is a wasm function within the same instance
         codePtr = *instance->calleeGroup()->entrypointLoadLocationFromFunctionIndexSpace(functionIndex);
         boxedCallee = CalleeBits::encodeNativeCallee(
             instance->calleeGroup()->wasmCalleeFromFunctionIndexSpace(functionIndex));
+        ASSERT(calleeStackSlot.unboxedInt64() == 0xBEEF);
+        calleeStackSlot = boxedCallee;
     }
 
-    Register& calleeStackSlot = partiallyConstructedCalleeFrame[static_cast<int>(CallFrameSlot::callee)];
-    ASSERT(calleeStackSlot.unboxedInt64() == 0xBEEF);
-    calleeStackSlot = boxedCallee;
```

Comment:
Since boxed wasm callee information on the stack is necessary for stack scanning, maybe, we should keep it.
How about retrieving functionIndex from boxed callee?